autonomous/autonomous.py

```diff
@@ -28,98 +48,85 @@ class AutoBase(AutonomousStateMachine):
 
     logger: logging.Logger
 
-    waypoints: List[Pose2d]
+    waypoints: List[Waypoint]
```

Comment:
I can't say I've been ever been happy with how we've done this. Could we move this into the constructor arguments?